src/withdraws/BaseWithdrawer.sol

```diff
@@ -0,0 +1,57 @@
+// SPDX-License-Identifier: BSD-3-Clause
+pragma solidity ^0.8.24;
+
+import {BaseStrategy} from "src/strategy/BaseStrategy.sol";
+import {AsyncWithdrawalLib} from "src/library/AsyncWithdrawalLib.sol";
+
+abstract contract BaseWithdrawer is BaseStrategy {
+    function initialize(
+        address admin,
+        string memory name,
+        string memory symbol,
+        uint8 decimals_,
+        bool countNativeAsset_,
+        bool alwaysComputeTotalAssets_
+    ) external virtual initializer {
+        _initialize(admin, name, symbol, decimals_, countNativeAsset_, alwaysComputeTotalAssets_);
+    }
+
+    function _initialize(
+        address admin,
+        string memory name,
+        string memory symbol,
+        uint8 decimals_,
+        bool countNativeAsset_,
+        bool alwaysComputeTotalAssets_
+    ) internal virtual {
+        __ERC20_init(name, symbol);
+        __AccessControl_init();
+        __ReentrancyGuard_init();
+        _grantRole(DEFAULT_ADMIN_ROLE, admin);
+
+        VaultStorage storage vaultStorage = _getVaultStorage();
+        vaultStorage.paused = true;
+        vaultStorage.decimals = decimals_;
+        vaultStorage.countNativeAsset = countNativeAsset_;
+        vaultStorage.alwaysComputeTotalAssets = alwaysComputeTotalAssets_;
+    }
+
+    function _feeOnRaw(uint256) public pure override returns (uint256) {
+        return 0;
+    }
+
+    function _feeOnTotal(uint256) public pure override returns (uint256) {
+        return 0;
+    }
+
+    function computeTotalAssets() public view virtual override returns (uint256 totalBaseBalance) {
+        totalBaseBalance = AsyncWithdrawalLib.computeTotalAssets();
```

Comment:
@danoctavian I'm not totally clear on how the lines for auditing are calculated. `AsyncWithdrawalLib` is pulled in here for `computeTotalAssets`. Should this override be in BaseWithdrawer or Withdrawer @danoctavian ?

I considered having the new `computeTotalAssets` logic inside `BaseWithdrawer` itself. 
The size different is 20,006 vs 19,788. Seems significant enough for it to be in the library itself.

Let me know what you think